src/gpu/d3d11/SDL_gpu_d3d11.c

```diff
@@ -3733,28 +3746,432 @@ static SDL_GpuCommandBuffer* D3D11_AcquireCommandBuffer(
+                            d3d11CommandBuffer->context,
+                            bindSlot->slot,
+                            1,
+                            &bufferContainer->activeBuffer->srv
+                        );
+                    }
+                    else
+                    {
+                        ID3D11DeviceContext_CSSetShaderResources(
+                            d3d11CommandBuffer->context,
+                            bindSlot->slot,
+                            1,
+                            &bufferContainer->activeBuffer->srv
+                        );
+                    }
+                }
+
+                D3D11_INTERNAL_TrackBuffer(
+                    d3d11CommandBuffer,
+                    bufferContainer->activeBuffer
+                );
+                break;
+
+            case SDL_GPU_RESOURCETYPE_STORAGE_BUFFER_READWRITE:
+                if (graphics)
+                {
+                    SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "D3D11 cannot bind read-write storage buffers in the graphics stage!");
+                }
+                else
+                {
+                    bufferContainer = (D3D11BufferContainer*) resourceBindings[i].resource.storageBufferReadWrite.gpuBuffer;
+
+                    D3D11_INTERNAL_PrepareGpuBufferForWrite(
+                        renderer,
+                        bufferContainer,
+                        resourceBindings[i].resource.storageBufferReadWrite.cycle
+                    );
+
+                    bindSlot = &resourceLayout->resourceSets[setIndex].resourceInfos[i].bindSlots[0];
+
+                    ID3D11DeviceContext_CSSetUnorderedAccessViews(
+                        d3d11CommandBuffer->context,
+                        bindSlot->slot,
+                        1,
+                        &bufferContainer->activeBuffer->uav,
+                        NULL
+                    );
+
+                    D3D11_INTERNAL_TrackBuffer(
+                        d3d11CommandBuffer,
+                        bufferContainer->activeBuffer
+                    );
+                }
+                break;
+
+            case SDL_GPU_RESOURCETYPE_STORAGE_TEXTURE_READONLY:
+                textureContainer = (D3D11TextureContainer*) resourceBindings[i].resource.storageTextureReadOnly.texture;
+                textureSubresource = D3D11_INTERNAL_FetchTextureSubresource(
+                    textureContainer->activeTexture,
+                    resourceBindings[i].resource.storageTextureReadOnly.layer,
+                    resourceBindings[i].resource.storageTextureReadOnly.mipLevel
+                );
+
+                for (j = 0; j < resourceLayout->resourceSets[setIndex].resourceInfos[i].bindSlotCount; j += 1)
+                {
+                    bindSlot = &resourceLayout->resourceSets[setIndex].resourceInfos[i].bindSlots[j];
+
+                    if (bindSlot->shaderStageFlag == SDL_GPU_SHADERSTAGE_VERTEX)
```

Comment:
it might make sense to extract this `XXSetShaderResource(ctx, stage, slot, handle)` pattern out into a helper since it's duplicated a lot, to reduce the risk of copy paste errors when this code is maintained later